### PR TITLE
Support multiple versions of one provider

### PR DESCRIPTION
**Pull request by jplewa**
## Task

Resolves: None

## Description

None

### CODE REVIEW

File: .github/workflows/publish_to_maven_local.yml [hunk 1]
```diff
@@ -25,8 +25,8 @@ jobs:
       fail-fast: false
       matrix:
         provider: [
-          cloudflare, slack, github, random, gcp, google-native, aws, aws-native, azure, azure-native, kubernetes,
-          nomad, docker, gitlab, digitalocean, alicloud
+          cloudflare4, cloudflare5, slack0, github5, random4, gcp6, google-native0, aws5, aws-native0, azure5,
+          azure-native1, kubernetes3, nomad0, docker3, docker4, gitlab4, gitlab5, gitlab6, digitalocean4, alicloud3
```
Comment (myhau) [inline .github/workflows/publish_to_maven_local.yml:29]: These numbers might be surprising for an unfamiliar reader. Have you tried doing it this way?

```yaml
provider: 
  - name: cloudflare
     majorVersion: 4
  - name: cloudflare
     majorVersion: 5
#  ...
```

I don't know if the above will work, but using `include` might be an option here

https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-adding-configurations

File: buildSrc/src/main/kotlin/SchemaMetadata.kt [hunk 2]
```diff
@@ -11,7 +12,10 @@ data class SchemaMetadata(
     val javaVersion: String,
     val javaGitTag: String,
     val customDependencies: List<String>,
-)
+) {
+    val versionedProvider: String
+        get() = "$providerName${javaVersion.split(".").first()}"
```
Comment (myhau) [inline buildSrc/src/main/kotlin/SchemaMetadata.kt:17]: This doesn't really change much, but shouldn't we use `kotlinVersion` here?

File: .github/workflows/prepare_release.yml [hunk 3]
```diff
@@ -56,10 +56,16 @@ jobs:
           import json
           with open("src/main/resources/version-config.json") as version_config:
             schemas = json.load(version_config)
-            version = [schema["kotlinVersion"] for schema in schemas if schema["providerName"] == "${{ matrix.provider }}"][0]
+            versions = [
+              schema["kotlinVersion"]
+              for schema
+              in schemas
+              if schema["providerName"] + schema["kotlinVersion"].split(".")[0]  == "${{ matrix.provider }}"
+            ]
```
Comment (myhau) [inline .github/workflows/prepare_release.yml:64]: (minor)

List comprehension formatted this way feels unnatural. Could you format it in IntelliJ and paste here?

**Review by myhau — APPROVED** [review]
Nice to have such support for multiple major versions :). 

On the other hand, while this whole machinery works nicely for our use-case, I keep thinking about community and how familiar all of this is / will be for them.

File: .github/workflows/publish_to_maven_local.yml (hunk 1 above)
Comment (jplewa) [reply .github/workflows/publish_to_maven_local.yml:29]: I created an issue for that: https://github.com/VirtuslabRnD/pulumi-kotlin/issues/294

**Pull request merged**